appengine/pond_storage/common.py

```diff
@@ -147,24 +147,67 @@ def get_user_ducks():
   duck_query = Duck.query(ancestor=user_key)
   duck_list = []
   for duck in duck_query:
-    duck_info = {'name': duck.name, 'duckUrl': duck.key.urlsafe()}
+    duck_info = {'name': duck.name, 'duck_key': duck.key.urlsafe()}
     if duck.published:
       duck_info['ranking'] = duck.leaderboard_entry_key.get().ranking
     duck_list.append(duck_info)
   return duck_list
 
-def get_duck_info(duck):
+def get_duck_info(duck, opt_keys=[]):
   """Returns object containing specified duck's information."""
-  duck_info = {
+  if opt_keys:
+    duck_info = {}
+    if 'name' in opt_keys:
+      duck_info['name'] = duck.name
+    if 'duck_key' in opt_keys:
+      duck_info['duck_key'] = duck.key.urlsafe()
+    if 'code' in opt_keys:
+      duck_info['code'] = {'js': duck.code.js, 'opt_xml': duck.code.opt_xml}
+    if 'published' in opt_keys:
+      duck_info['published'] = duck.published
+  else:
+    duck_info = {
       'name': duck.name,
       'duck_key': duck.key.urlsafe(),
       'code': {'js': duck.code.js, 'opt_xml': duck.code.opt_xml},
-      'published': duck.published,
-  }
+      'published': duck.published
+    }
   if duck.published:
     duck_info['ranking'] = duck.leaderboard_entry_key.get().ranking
+
   return duck_info
 
+def entries_to_duck_info(entries):
```

Comment:
I would prefer if the `duck_key` filter was passed into this method from `get_top_ducks`.

Reply:
done